src/samarium/python.py

```diff
@@ -0,0 +1,73 @@
+from collections.abc import Callable
+from typing import Any, Generic, TypeVar
+from functools import wraps
+from samarium.classes import (
+    NULL,
+    Array,
+    Integer,
+    Null,
+    String,
+    Table,
+)
+from samarium.classes.base import Attrs
+
+F = TypeVar("F", bound=Callable)
+
+
+class PythonExport(Generic[F]):
+    def __init__(self, o: F) -> None:
+        self.o = o
+
+
+def export(func: F) -> PythonExport[F]:
+    "Marks an object to be exported to Samarium"
+    return PythonExport(func)
+
+
+def to_python(obj: Attrs) -> object:
+    if isinstance(obj, (String, Integer)):
+        return obj.val
+    elif isinstance(obj, Null):
+        return None
+    elif isinstance(obj, Array):
+        return [to_python(i) for i in obj.val]
+    elif isinstance(obj, Table):
+        return {to_python(k): to_python(v) for k, v in obj.val.items()}
+
+
+def to_samarium(obj: object):
+    if isinstance(obj, (int, bool)):
+        return Integer(obj)
+    elif isinstance(obj, str):
+        return String(obj)
+    elif obj is None:
+        return NULL
+    elif isinstance(obj, (list, tuple, set)):
+        return Array([to_samarium(i) for i in obj])
+    elif isinstance(obj, dict):
+        return Table({to_samarium(k): to_samarium(v) for k, v in obj.items()})
+
+
+def function(func):
+    """Wraps a Python function to be used in Samarium"""
+    @wraps(func)
+    def wrapper(*_args):
+        args = (to_python(arg) for arg in _args)
+        return to_samarium(func(*args))
+
+    return wrapper
+
+
+class SmProxy(Attrs):
+
+    def __init__(self, v: Any):
+        self.v = v
+
+    def __getattr__(self, name: str):  # type: ignore
+        if name.startswith("sm_"):
+            attr = getattr(self.v, name.removeprefix("sm_"))
+
+            if callable(attr):
+                return function(attr)  # type: ignore
```

Comment:
Won't this also trigger for stuff like classes or other callable objects?

Reply:
> Won't this also trigger for stuff like classes or other callable objects?

Yes, will fix. 
I think I'll add a `__call__` on `SmProxy` and wrap it in that rather than in `function`